bin/all_sky_search/pycbc_calculate_dq

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+""" Ingest and process DQ data
+"""
+import logging, argparse, numpy, h5py, multiprocessing, time, copy
+from six.moves import (range, zip_longest)
+import pycbc, pycbc.strain
+from pycbc.version import git_verbose_msg as version
+from pycbc.fft.fftw import set_measure_level
+from pycbc.workflow import resolve_td_option
+from ligo.segments import segmentlist, segment
+set_measure_level(0)
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--version', action='version', version=version)
+parser.add_argument('--verbose', action="store_true")
+parser.add_argument("--dq-sample-rate", default=1, type=int,
+                    help="Sample rate to calculate dq timeseries, "
+                         "in Hz [default=1Hz]")
+parser.add_argument("--dq-sample-method", required=True,
+                    help="Method used to calculate dq timeseries. "
+                         "Options are 'max' or 'blrms'")
+parser.add_argument("--gps-start-time", type=int,required=True,
+                    help="The gps start time of the data "
+                          "(integer seconds)")
+parser.add_argument("--gps-end-time", type=int,required=True,
+                    help="The gps end time of the data "
+                         "(integer seconds)")
+parser.add_argument("--output-file", required=True,
+                    help="name of output file")
+
+pycbc.strain.insert_strain_option_group(parser, gps_times=False)
+
+parser.set_defaults(pad_data=0)
+
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
+def get_dq(seg):
+    """ Get the DQ data for the given data chunck
+    """
+    
+    logging.info('Getting dq values for %.1f-%.1f (%.1f s)', seg[0],
+                 seg[1], abs(seg[1]-seg[0]))
+    argstmp = copy.deepcopy(args)
+    argstmp.gps_start_time = int(seg[0])
+    argstmp.gps_end_time = int(seg[1])
+    tmp_segment = segment([argstmp.gps_start_time, argstmp.gps_end_time])
+    argstmp.channel_name = resolve_td_option(args.channel_name, tmp_segment)
+
+    try:
+        gwstrain = pycbc.strain.from_cli(argstmp)
+    except RuntimeError:
+        time.sleep(10)
+        return get_dq(seg)
```

Comment:
I think this `try` block warrants a comment to explain why it is needed.

Reply:
This is copied over from `pycbc_calcuclate_psd`:
https://github.com/gwastro/pycbc/blob/90503e2bf6ec60715b3b04139d7c77f4d6e05046/bin/all_sky_search/pycbc_calculate_psd#L53-L59
I'm happy to also note the reason here. But more generally, do you think this try statement is needed?

Reply:
Hmm, if we think a retry mechanism is beneficial, why not put it in `pycbc.strain.from_cli()` itself? Otherwise we need to duplicate this `try`-`except` code every time we want to read some data.

Reply:
@ahnitz originally wrote this part of `pycbc_calculate_psd` - perhaps this try statement is no longer necessary in general? I'm open to moving/deleting this part of the code.

Reply:
I suggest removing the try-except block from here. If we run into I/O issues that warrant putting it back, then we look for a solution at the level of the `pycbc.strain` module.

Reply:
I've now removed this try block. I agree that makes sense in the short term.